test/test_console_tool.py

```diff
@@ -121,13 +127,72 @@ def test_buckets(self):
         {
             "accountId": "my-account",
             "bucketId": "bucket_1",
+            "bucketInfo": {},
             "bucketName": "your-bucket",
-            "bucketType": "allPrivate"
+            "bucketType": "allPrivate",
+            "lifecycleRules": [],
+            "revision": 1
         }
         '''
 
         self._run_command(['delete_bucket', 'your-bucket'], expected_stdout, '', 0)
 
+    def test_bucket_info_from_file(self):
+
+        self._authorize_account()
+        self._run_command(['create_bucket', 'my-bucket', 'allPublic'], 'bucket_0\n', '', 0)
+
+        with TempDir() as temp_dir:
+            bucket_info = {'color': 'blue'}
+            bucket_info_file = self._make_local_file(temp_dir, 'bucket_info.json')
+            with open(bucket_info_file, 'wb') as f:
+                f.write(six.b(json.dumps(bucket_info)))
+
+            expected_stdout = '''
+                {
+                    "accountId": "my-account",
+                    "bucketId": "bucket_0",
+                    "bucketInfo": {
+                        "color": "blue"
+                    },
+                    "bucketName": "my-bucket",
+                    "bucketType": "allPrivate",
+                    "lifecycleRules": [],
+                    "revision": 2
+                }
+                '''
+            self._run_command(
+                [
+                    'update_bucket', '--bucketInfo', '@' + bucket_info_file, 'my-bucket',
+                    'allPrivate'
+                ], expected_stdout, '', 0
+            )
+
+    def test_bucket_info_from_json(self):
+
+        self._authorize_account()
+        self._run_command(['create_bucket', 'my-bucket', 'allPublic'], 'bucket_0\n', '', 0)
+
+        bucket_info = {'color': 'blue'}
+
+        expected_stdout = '''
+            {
+                "accountId": "my-account",
+                "bucketId": "bucket_0",
+                "bucketInfo": {
+                    "color": "blue"
+                },
+                "bucketName": "my-bucket",
+                "bucketType": "allPrivate",
+                "lifecycleRules": [],
+                "revision": 2
+            }
+            '''
+        self._run_command(
+            ['update_bucket', '--bucketInfo', json.dumps(bucket_info), 'my-bucket', 'allPrivate'],
+            expected_stdout, '', 0
+        )
+
```

Comment:
There are no tests for failures. The test that checks whether the Conflict exception works correctly is especially important as we do know `b2_fuse` project will build a hard consistency model over it.